### DISCUSSION TITLE
installing pip dependencies stuck

### DISCUSSION THREAD

**Discussion by erhb9** [hlky/stable-diffusion#259] [Q&A] [answered]
Why do I have to reinstall env folder each time I want to open the webUI? 
When I close WebUI and want to re-open it the pip install dependencies stucks and the python application CPU usage wents from 0.1% to 12%. And to resolve it I have reinstall it everytime I want to open the WebUI. But can someone help me to resolve it without deleting the env folder each time?

**Comment by ghost** [comment, marked as answer]
Edit: Try
`activate ldo`
`python scripts/relauncher.py`
webui.cmd always seems to redownload everything, not sure why it's working that way. Maybe I'll take a look and make a pull request.

**Reply by erhb9** [reply]
relauncher.py is stuck too now
Edit: I re opened it and it worked. thanks

**Reply by y0himba** [reply]
Windows10, c:\stable-diffusion, running miniconda3 prompt as admin.

I followed the process above, relauncher.py looped each time I ran it with:

`Relauncher: Launching...
Traceback (most recent call last):
  File "scripts/webui.py", line 3, in <module>
    from frontend.frontend import draw_gradio_ui
ModuleNotFoundError: No module named 'frontend'
Relauncher: Process is ending. Relaunching in 1s...`

webui.cmd and webuildm.cmd both sat eternally at "Installing pip dependencies:" with the spinning indicator.

No idea what's going on or why.  Never had an issue installing dependencies or packages before this one.

Thanks for any help.  I'm dying to try this.

**Reply by ghost** [reply]
Hey @y0himba I ran into this a while back, the problem stemmed from pip getting confused with the >= signs in the root `environment.yaml` file

I think newer versions of pip support that feature. Try:
`python -m pip install --upgrade pip`

in your conda environment and then run whichever .cmd file you use again, if it still gets stuck for longer than a couple minutes you could try doing a search and replace `>=` to `==` in the `environment.yaml` file and running the .cmd file one last time.

**Reply by y0himba** [reply]
Upgraded pip to 22.2.2
Changed `>=` to `==`
Still installing pip dependencies infinitely

I appreciate the answer though, equaling the version numbers I really thought would work...

EDIT:  I tried removing the pip dependencies from `environment.yaml`. That resulted in `taming-transformers` dependency not being found, then looping `relauncher.py` 

`Relauncher: Launching...
Traceback (most recent call last):
  File "scripts/webui.py", line 3, in <module>
    from frontend.frontend import draw_gradio_ui
ModuleNotFoundError: No module named 'frontend'
Relauncher: Process is ending. Relaunching in 1s...`

**Reply by ghost** [reply]
What happens if you do:

`pip install test-tube>=0.7.5`

Also try:
`pip3 install test-tube>=0.7.5`

Any difference?

**Reply by ericmdillon** [reply]
I've been stuck on this same problem with the exact same results as @y0himba, and I found a solution that seems to work for me so far:
1. Upgrade pip with `python -m pip install --upgrade pip`
2. `pip install test-tube>=0.7.5`
3. Change your `environment.yaml` file to state the following:
```
name: ldm
channels:
  - pytorch
  - defaults
dependencies:
  - git
  - python=3.8.5
  - pip=20.3
  - cudatoolkit=11.3
  - pytorch=1.11.0
  - torchvision=0.12.0
  - numpy=1.19.2
  - pip:
    - -e git+https://github.com/CompVis/taming-transformers#egg=taming-transformers
    - -e git+https://github.com/hlky/k-diffusion-sd#egg=k_diffusion
```
4. Run webui.cmd.

I've not extensively tested this yet, but it does seem to work enough to generate a text2img prompt. Hope it helps!

**Reply by ghost** [reply]
Take a look at this guys:
https://github.com/hlky/stable-diffusion-webui/issues/585

A useful technique was https://github.com/hlky/stable-diffusion-webui/issues/585#issuecomment-1236307184

**Reply by y0himba** [reply]
`(ldm) C:\stable-diffusion>pip install test-tube>=0.7.5

(ldm) C:\stable-diffusion>`

It sat for about 60 seconds and then went back to the prompt, so I am assuming it completed...I am trying ericmdillon's answer now...

**Reply by y0himba** [reply]
eric's solution above got pip installed for me, it completed.  However, `relauncher.py` now lopps with the following output:
`Retrieving notices: ...working... done
Relauncher: Launching...
Traceback (most recent call last):
  File "scripts/webui.py", line 3, in <module>
    from frontend.frontend import draw_gradio_ui
  File "c:\stable-diffusion\frontend\frontend.py", line 1, in <module>
    import gradio as gr
ModuleNotFoundError: No module named 'gradio'
Relauncher: Process is ending. Relaunching in 1s...`

**Reply by ghost** [reply]
You need to take about 5-10 minutes and manually install all the packages from environment.yaml file. I had to do the same thing, must be a windows issue. Make sure you're in your conda environment before running these. It should not say (base) before your commands, it should say (ldm) or similar

`pip install albumentations==0.4.3`
`pip install opencv-python==4.1.2.30`
`pip install opencv-python-headless==4.1.2.30`
`pip install pudb==2019.2`
`pip install imageio==2.9.0`
`pip install imageio-ffmpeg==0.4.2`
`pip install pytorch-lightning==1.4.2`
`pip install omegaconf==2.1.1`
`pip install test-tube==0.7.5`
`pip install streamlit==0.73.1`
`pip install einops==0.3.0`
`pip install torch-fidelity==0.3.0`
`pip install transformers==4.19.2`
`pip install torchmetrics==0.6.0`
`pip install kornia==0.6`
`pip install gradio==3.1.6`
`pip install accelerate==0.12.0`
`pip install pynvml==11.4.1`
`pip install basicsr==1.3.4.0`
`pip install facexlib==0.2.3`
`pip install python-slugify==6.1.2`
`pip install streamlit==1.12.2`
`pip install -e git+https://github.com/CompVis/taming-transformers#egg=taming-transformers`
`pip install -e git+https://github.com/openai/CLIP#egg=clip`
`pip install -e git+https://github.com/TencentARC/GFPGAN#egg=GFPGAN`
`pip install -e git+https://github.com/xinntao/Real-ESRGAN#egg=realesrgan`
`pip install -e git+https://github.com/hlky/k-diffusion-sd#egg=k_diffusion`
`pip install -e .`

**Reply by y0himba** [reply]
Finally got it. the solution offered above by @cherrerajobs did it.  Thanks for the help folks! I now have a fancy corgi wearing a top hat styled like an oil painting.

**Reply by y0himba** [reply]
Next issue:  `LDSR not found at path, please make sure you have cloned the LDSR repo to ./src/latent-diffusion/`

I am not familiar with LSDR, I found `https://github.com/CompVis/latent-diffusion` but I am not sure this is what I am looking for?

If it is, then I use `git clone https://github.com/CompVis/latent-diffusion` into` /src/latent-diffusion`, correct?

EDIT: I have figured out that I need the LSDR models, where do I get them from?  Are they the models linked on https://github.com/CompVis/latent-diffusion ?

**Reply by ghost** [reply]
Yup! Then in there you can run download_model.bat to download the model. Those instructions worked well for me, it picked it right up and enabled it in the interface.

**Reply by y0himba** [reply]
No batch files were included in my clone of the repo, however there are two shell command scripts.  When I run the shell command scripts, they don't download anything...

**Reply by ghost** [reply]
Follow the instructions here:
https://github.com/hlky/stable-diffusion-webui/blob/master/README.md#ldsr
